### ISSUE TITLE
[BUG] Criteria applied to CActiveRecord::getRelated does not merge correctly

### ISSUE THREAD

**Issue by davidhirtz** [yiisoft/yii#3579]
I have traced it back to [CActiveFinder.php#L266](https://github.com/yiisoft/yii/blob/master/framework/db/ar/CActiveFinder.php#L266), but I am unable to solve this bug. 

Simple example below:

Comment.php

``` php
public function testScope()
{
    return $this->with(array(
        'user'=>array(
            'select'=>array('username'),
        ),
    ));
}
```

Controller.php

``` php
Comment::model()->findAll(array(
    'with'=>array(
        'user'=>array(
            'select'=>array('id'),
        ),
    ),
    'scopes'=>array('testScope'),
));
```

Above works as expected selecting both "id" and "username" from Comment relation "user".

``` php
$parentModel->getRelated('comments', true, array(
    'with'=>array(
        'user'=>array(
            'select'=>array('id'),
        ),
    ),
    'scopes'=>array('testScope'),
));
```

This should work too, but it doesn't, it only selects user's "id" because "testScope" is overwritten by the "with" declaration. 

Thank you for your feedback!

**Comment by claudiocarotenuto** [comment]
I made a pull request to solve this bug.
In my solution I simply took the code from [CDbCriteria::mergeWith](https://github.com/yiisoft/yii/blob/master/framework/db/schema/CDbCriteria.php#L600) (lines from 600 to 631) and added it to [CActiveRelation::mergeWith](https://github.com/yiisoft/yii/blob/master/framework/db/ar/CActiveRecord.php#L2213) so that existing with criteria is not overwritten by criteria one.

**Comment by samdark** [comment]
Won't be fixed in 1.1.
